hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java

```diff
@@ -996,6 +996,22 @@ public void run() {
     @Override
     public void run() {
       while (true) {
+        if (futures.isEmpty()) {
+          synchronized (this) {
+            try {
+              // waiting for tokenRenewerThreadTimeout milliseconds
+              long waitingTimeMs = Math.min(10000, Math.max(500, tokenRenewerThreadTimeout));
```

Comment:
500, 10000 Should we increase the configuration？

Reply:
@slfan1989 what do you suggest increasing them to?